src/plugins/flattenModifier.ts

```diff
@@ -52,8 +68,167 @@ export class FlattenModifier {
+                    (property.schema.extensions = property.schema.extensions || {})[hasBeenFlattened] = true;
+                    NodeCliHelper.setCliFlattened(property, true);
+                }
+            }
+        }
+
+        schema.extensions[isCurrentlyFlattening] = false;
+    }
+
+    private flattenPayload(flattenMultiPayload: boolean, propNumThreshold: number): void {
+        for (const group of this.session.model.operationGroups) {
+            for (const operation of group.operations) {
+                // when there are multiple requests in an operation
+                // and the generator asks not to flatten them
+                if (length(operation.requests) > 1 && !flattenMultiPayload) {
+                    continue;
+                }
+
+                for (const request of values(operation.requests)) {
+                    const body = values(request.parameters).first(p => p.protocol.http?.in === ParameterLocation.Body && p.implementation === ImplementationLocation.Method);
+
+                    if (body && isObjectSchema(body.schema)) {
+                        const schema = <ObjectSchema>body.schema;
+                        if (schema.discriminator) {
+                            // skip flattening on polymorphic payloads, since you don't know the actual type.
+                            continue;
+                        }
+
+                        let flattenOperationPayload = NodeCliHelper.isCliFlatten(body) && !NodeCliHelper.isCliFlattened(body);
+                        if (!flattenOperationPayload) {
+                            // told not to explicitly.
+                            continue;
+                        }
+
+
+                        if (!flattenOperationPayload) {
+                            // get the count of the (non-readonly) properties in the schema
+                            flattenOperationPayload = length(values(getAllProperties(schema)).where(property => property.readOnly !== true && property.schema.type !== SchemaType.Constant)) <= propNumThreshold;
+                        }
+
+                        if (flattenOperationPayload) {
+                            // For backward compatiable. Follow flatten payload logic in modelerfour .
+                            FlattenHelper.flattenParameter(request, body, [], '', true);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private flattenOperationParameter(polyEnabled: boolean): void {
+        if (polyEnabled) {
+            Helper.enumrateOperationGroups(this.session.model.operationGroups, [], (desc) => this.flattenPolyOperationParameter(desc), CliCommonSchema.CodeModel.NodeTypeFlag.operation);
+        }
+        Helper.enumrateOperationGroups(this.session.model.operationGroups, [], (desc) => this.flattenNormalOperationParameters(desc), CliCommonSchema.CodeModel.NodeTypeFlag.operation);
+    }
+
+    private removeUnusedModel(): void {
```

Comment:
>removeUnusedModel [](start = 12, length = 17)

what's this for?

Reply:
it's for `cli-flatten-keep-unused-flattened-models`. To keep same logic as modelerfour